metrics/prometheus.go

```diff
@@ -0,0 +1,172 @@
+package metrics
+
+import (
+	"github.com/prebid/prebid-cache/config"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+/**************************************************
+ *	Still missing:
+ **************************************************
+ *	> Register methods for Prometheus metrics in metrics/prometheus.go
+ *	> Modify all histograms
+ *	> Tests for config.go
+ *	> Tests for metrics/prometheus.go
+ *	> Upload so they can see progress
+ *	> Make the option betewwn Prometheus and InfluxDB configurable
+ **************************************************/
+/**************************************************
+ *	Object definition
+ **************************************************/
+
+type PrometheusMetrics struct {
+	Registry        *prometheus.Registry
+	Puts            *PrometheusMetricsEntry
+	Gets            *PrometheusMetricsEntry
+	PutsBackend     *PrometheusMetricsEntryByFormat
+	GetsBackend     *PrometheusMetricsEntry
+	Connections     *PrometheusConnectionMetrics
+	ExtraTTLSeconds *prometheus.HistogramVec
+}
+type PrometheusMetricsEntry struct {
+	Duration   *prometheus.HistogramVec
+	Errors     prometheus.Counter
```

Comment:
Why three counters rather than a vector with labels?

Reply:
I rewrote the `PrometheusMetrics` object in a way I believe better adapts to the Prometheus metric paradigms. Please let me know what you think of this new approach. I can always roll back.